lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2355,7 +2356,9 @@ void IRGenDebugInfoImpl::handleFragmentDIExpr(
       emitPhysicalStructMemberFixedOffset(IGM, ParentSILType, VD);
   llvm::Type *FieldTy =
       getPhysicalStructFieldTypeInfo(IGM, ParentSILType, VD)->getStorageType();
-  assert(Offset && FieldTy && "Non-fixed type?");
+  // Doesn't support non-fixed type right now
+  if (!Offset || !FieldTy)
+    return false;
```

Comment:
I'd be curious to see an example of the IR here. Even for a non-fixed size type, I would expect us to emit some sort of fixed-size container that optionally contains a pointer to a heap/stack-allocated memory, and I would expect SROA to only split apart that fixed-size container and not the variable-sized part. We need to fix this right now, but maybe keep this in mind.